crates/sequencer/src/sequencer.rs

```diff
@@ -25,3 +36,173 @@ where
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::File;
+    use std::sync::Arc;
+
+    use blockifier::abi::abi_utils::get_storage_var_address;
+    use blockifier::execution::contract_class::{ContractClass, ContractClassV0};
+    use blockifier::state::state_api::State as BlockifierState;
+    use blockifier::transaction::account_transaction::AccountTransaction;
+    use starknet::core::types::FieldElement;
+    use starknet_api::core::{ChainId, ClassHash, ContractAddress, Nonce};
+    use starknet_api::hash::StarkFelt;
+    use starknet_api::transaction::{
+        Calldata, Fee, InvokeTransaction, InvokeTransactionV1, TransactionHash,
+        TransactionSignature,
+    };
+
+    use crate::constants::test_constants::{
+        FEE_TOKEN_ADDRESS, ONE_BLOCK_NUMBER, ONE_BLOCK_TIMESTAMP, ONE_CLASS_HASH,
+        SEQUENCER_ADDRESS, TEST_ADDRESS, TEST_CONTRACT_ACCOUNT, TEST_CONTRACT_ADDRESS,
+        TWO_CLASS_HASH, ZERO_FELT,
+    };
+    use crate::state::State;
+
+    use super::*;
+
+    fn read_contract_class(path: &str) -> ContractClass {
+        let reader = File::open(path).unwrap();
+        let contract_class: ContractClassV0 = serde_json::from_reader(reader).unwrap();
+
+        ContractClass::V0(contract_class)
+    }
+
+    fn declare_and_deploy_contract(
+        path: &str,
+        address: ContractAddress,
+        class_hash: ClassHash,
+        mut state: &mut State,
+    ) {
+        let contract_class = read_contract_class(path);
+
+        state
+            .set_contract_class(&class_hash, contract_class)
+            .unwrap();
+        state.set_class_hash_at(address, class_hash).unwrap();
+    }
+
+    fn fund(address: StarkFelt, mut state: &mut State) {
+        state.set_storage_at(
+            *FEE_TOKEN_ADDRESS,
+            get_storage_var_address("ERC20_balances", &[address]).unwrap(),
+            StarkFelt::from(u128::MAX),
+        );
+    }
+
+    #[test]
+    fn test_sequencer() {
+        // Given
+        let mut state = State::default();
+        let mutable = &mut state;
+
+        declare_and_deploy_contract(
+            "src/test_data/compiled_classes/counter.json",
+            *TEST_CONTRACT_ADDRESS,
+            *ONE_CLASS_HASH,
+            mutable,
+        );
+        declare_and_deploy_contract(
+            "src/test_data/compiled_classes/account.json",
+            *TEST_CONTRACT_ACCOUNT,
+            *TWO_CLASS_HASH,
+            mutable,
+        );
+        fund(*TEST_ADDRESS, mutable);
+
+        let context = BlockContext {
+            chain_id: ChainId("KKRT".into()),
+            block_number: *ONE_BLOCK_NUMBER,
+            block_timestamp: *ONE_BLOCK_TIMESTAMP,
+            sequencer_address: *SEQUENCER_ADDRESS,
+            fee_token_address: *FEE_TOKEN_ADDRESS,
+
+            vm_resource_fee_cost: Arc::new(
```

Comment:
what is the impact of setting n_steps == 1_f64?
why f64?

i understnad its from blockifier but for my culture

Reply:
I'm not 100% sure of what I am saying, but I think this mapping translates ALL the costs in term of cairo gas. So this maps builtins and steps to a single unit of reference, which I believe in blockifier thel call gas.

Reply:
can we verify and add a comment?